base_view_inheritance_extension/models/ir_ui_view.py

```diff
@@ -41,12 +41,24 @@ def __missing__(self, key):
 class IrUiView(models.Model):
     _inherit = 'ir.ui.view'
 
+    user_ids = fields.Many2many('res.users', string='Restrict to users')
+
     @api.model
     def apply_inheritance_specs(self, source, specs_tree, inherit_id):
         for specs, handled_by in self._iter_inheritance_specs(specs_tree):
             source = handled_by(source, specs, inherit_id)
         return source
 
+    @api.model
+    def get_inheriting_views_arch(self, view_id, model):
+        return [
+            (arch, view_id_)
+            for arch, view_id_ in
+            super(IrUiView, self).get_inheriting_views_arch(view_id, model)
+            if not self.sudo().browse(view_id_).user_ids or
+            self.sudo().browse(view_id_).user_ids & self.env.user
```

Comment:
means return this if does not have user_ids restriction or if the exclusive union (intersection) of user ids is and current user  is not null.  compact but hard to read . if there aren't computational advantages i do not know about ,  i believe this is less readable. gust a general opinion, MR looks good, had to write *something* on review.